db/database.go

```diff
@@ -1161,147 +1162,151 @@ func (db *Database) UpdateAllDocChannels(regenerateSequences bool) (int, error)
 
 		var importRow QueryIdRow
 		for results.Next(&importRow) {
-			if db.ResyncManager.ShouldStop() {
+			select {
```

Comment:
Instead of nesting all the processing inside the 'default', I might just treat this as a check on the terminator channel prior to continuing, like:
```
select {
    case <-terminator:
         clean up and exit
    default:
}
[we now continue our regular programming]
```